.github/workflows/publish.yml

```diff
@@ -9,7 +9,38 @@ permissions:
   contents: write
 
 jobs:
+  version:
+    name: Update version
+    runs-on: ubuntu-latest
+    if: startsWith(github.ref, 'refs/tags/v')
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.12
+      - name: Install poetry
+        run: |
+          pip install --upgrade poetry
+      - name: Update version
+        run: |
+          VERSION=${GITHUB_REF#refs/tags/v}
+          poetry version $VERSION
+      - name: Commit changes
+        run: |
+          git config --local user.email "${{ secrets.GIT_USER_EMAIL }}"
+          git config --local user.name "${{ secrets.GIT_USER_NAME }}"
+          git commit -am "chore: Update version"
+
+      - name: Push changes
```

Comment:
@shemogumbe why have you reverted the suggested change? can you provide context when you do things like this please?

Reply:
The idea here is to make a commit with the version change, then push that commit.
The change  was up to the stage of the commit, changes to allow a git push are new.
I think the commit is factored in the latest changes, right?

Reply:
but why aren't you simply using a git push command in the previous script block?

Reply:
so am I understanding things correctly that you'll revert this change to: remove the push action step (block below) and add a git push command to the block above?

Reply:
Sure, on it